src/librbd/cache/ImageCache.h

```diff
@@ -6,21 +6,28 @@
 
 #include "include/buffer_fwd.h"
 #include "include/int_types.h"
+#include "librbd/io/Types.h"
 #include <vector>
 
 class Context;
 
 namespace librbd {
+struct ImageCtx;
 namespace cache {
 
 /**
  * client-side, image extent cache interface
  */
-struct ImageCache {
-  typedef std::vector<std::pair<uint64_t,uint64_t> > Extents;
+template <typename ImageCtxT = ImageCtx>
+class ImageCache {
+protected:
+  ImageCache() {}
+public:
+  /* Extent: offset in bytes, length in bytes */
+  typedef std::pair<uint64_t,uint64_t> Extent;
```

Comment:
Nit: perhaps move (reuse) these to (from) `librbd/io/Types.h'

Reply:
That sounds reasonable. There were some confusing and subtly different Extent types back when I started this. That seems to have simplified.

Reply:
Done